<?xml version='1.0' encoding='utf-8'?>

<launch>
  <!--  激光雷达相对于底盘的位置  -->
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser4" 
    args="0.12 0.0 0.15 0 0 0.0 /base_link /laser_frame 40" />
    
  <!-- 导入建好的地图 -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find auto_demo)/maps/automap.yaml" /> 

  <!-- 机器人在地图中的初始化位置 -->
  <arg name="initial_pose_x"  default="0.069"/>
  <arg name="initial_pose_y"  default="-0.020"/>
  <arg name="initial_pose_a"  default="3.014"/>

  <!-- 启动amcl节点 --> 
  <node pkg="amcl" type="amcl" name="amcl" output="screen">
  <remap from="scan" to="scan"/>
    <param name="use_map_topic"             value="false"/>
    <!-- Publish scans from best pose at a max of 10 Hz -->
    <!-- Publish scans from best pose at a max of 10 Hz -->
  <param name="odom_model_type" value="omni"/>
  <param name="odom_alpha5" value="0.1"/>
  <param name="transform_tolerance" value="0.2" />
  <param name="gui_publish_rate" value="10.0"/>
  <param name="laser_max_beams" value="30"/>
  <param name="min_particles" value="500"/>
  <param name="max_particles" value="5000"/>
  <param name="kld_err" value="0.05"/>
  <param name="kld_z" value="0.99"/>
  <param name="odom_alpha1" value="0.2"/>
  <param name="odom_alpha2" value="0.2"/>
  <!-- translation std dev, m -->
  <param name="odom_alpha3" value="0.8"/>
  <param name="odom_alpha4" value="0.2"/>
  <param name="laser_z_hit" value="0.5"/>
  <param name="laser_z_short" value="0.05"/>
  <param name="laser_z_max" value="0.05"/>
  <param name="laser_z_rand" value="0.5"/>
  <param name="laser_sigma_hit" value="0.2"/>
  <param name="laser_lambda_short" value="0.1"/>
  <param name="laser_lambda_short" value="0.1"/>
  <param name="laser_model_type" value="likelihood_field"/>
  <!-- <param name="laser_model_type" value="beam"/> -->
  <param name="laser_likelihood_max_dist" value="2.0"/>
  <param name="update_min_d" value="0.1"/>
  <param name="update_min_a" value="0.5"/>
  <param name="odom_frame_id" value="odom"/>
  <param name="base_frame_id" value="base_link"/>
  <param name="resample_interval" value="1"/>
  <param name="recovery_alpha_slow" value="0.0"/>
  <param name="recovery_alpha_fast" value="0.0"/>
  <param name="initial_pose_x"      value="$(arg initial_pose_x)"/>
  <param name="initial_pose_y"      value="$(arg initial_pose_y)"/>
  <param name="initial_pose_a"      value="$(arg initial_pose_a)"/>

  </node>
 
  <!-- 启动move_base节点 -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="allow_unknown" value="false" />
    <param name="shutdown_costmaps" value="false" />
    <param name="controller_frequency" value="5.0" /> <!--原来参数为5-->
    <param name="controller_patience" value="3.0" /> <!--原来参数为3-->
    <param name="planner_patience" value="5.0" />
    <param name="controller_patience" value="3.0" />
    <param name="conservative_reset_dist" value="3.0" />
    <param name="planner_frequency" value="1.0" />
    <param name="oscillation_timeout" value="10.0" />
    <param name="oscillation_distance" value="0.22" />

    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />
    <rosparam file="$(find auto_demo)/params/news_v1/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find auto_demo)/params/news_v1/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find auto_demo)/params/news_v1/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find auto_demo)/params/news_v1/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find auto_demo)/params/news_v1/global_planner_params.yaml" command="load" />
    <rosparam file="$(find auto_demo)/params/news_v1/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find auto_demo)/params/news_v1/base_local_planner_params.yaml" command="load" />

  </node>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find auto_demo)/rviz/slam.rviz" required="true" />

</launch>
